src/main/scala/intellij/haskell/module/HaskellModuleBuilder.scala

```diff
@@ -218,12 +223,18 @@ object HaskellModuleBuilder {
     }
   }
 
-  private def downloadHaskellPackageSources(project: Project, stackPath: String, haskellPackages: Seq[HaskellPackageInfo], progressIndicator: ProgressIndicator) = {
+  private def downloadHaskellPackageSources(project: Project, libRoot: File, stackPath: String, haskellPackages: Seq[HaskellPackageInfo], progressIndicator: ProgressIndicator) = {
     val step = 0.8 / haskellPackages.size
     var progressFraction = InitialProgressStep
-    haskellPackages.flatMap { packageInfo =>
-      val fullName = packageInfo.name + "-" + packageInfo.version
-      val stdErr = CommandLine.runProgram(Some(project), project.getBasePath + File.separator + LibName, stackPath, Seq("unpack", fullName), 10000, captureOutputToLog = true, logErrorAsInfo = true).map(_.getStderr)
+
+    libRoot.listFiles().filterNot(subDir => haskellPackages.map(_.dirName).contains(subDir.getName)).foreach(FileUtil.delete)
+
+    haskellPackages.filter(packageInfo => {
```

Comment:
Hi @rikvdkleij, just add some screenshot to make it more clear.

If we only use the filterNot part as you said

```
haskellPackages.filter(packageInfo => {
      !new File(project.getBasePath + File.separator + LibName + File.separator + packageInfo.dirName).exists()
    }).flatMap { packageInfo =>
      val stdErr = CommandLine.runProgram(Some(project), project.getBasePath + File.separator + LibName, stackPath, Seq("unpack", packageInfo.dirName), 10000, captureOutputToLog = true, logErrorAsInfo = true).map(_.getStderr)
      progressFraction = progressFraction + step
      progressIndicator.setFraction(progressFraction)

      if (stdErr.exists(_.contains("not found"))) {
        Seq()
      } else {
        Seq(packageInfo)
      }
    }
```

Those packages which are already downloaded will never be displayed on the module dependencies panel.

<img width="1058" alt="screen shot 2017-02-25 at 2 06 09 am" src="https://cloud.githubusercontent.com/assets/6234553/23314965/047e92d0-faff-11e6-97a3-854750b17656.png">

If we do what I just did in this commit, all the packages no matter it is already downloaded will be kindly displayed on the module dependencies panel.

<img width="1059" alt="screen shot 2017-02-25 at 2 08 57 am" src="https://cloud.githubusercontent.com/assets/6234553/23315044/69d8c5a6-faff-11e6-8463-f9d2d5747b83.png">